detox/test/e2e/15.urls.test.js

```diff
@@ -1,23 +1,29 @@
 describe('Open URLs', () => {
 
-  it('device.launchApp({{newInstance: true, url: url}) should launch app and trigger handling open url handling in app', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true, url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  const testUrl = 'detoxtesturlscheme://such-string';
+
+  it('device.launchApp({{newInstance: true, testUrl: testUrl}) should launch app and trigger handling open testUrl handling in app', async () => {
+    await device.launchApp({newInstance: true, url: testUrl});
+    await expect(element(by.text(testUrl))).toBeVisible();
   });
 
-  it('device.openURL({url: url}) should trigger open url handling in app when app is in foreground', async () => {
-    const url = 'detoxtesturlscheme://such-string';
+  it('device.openURL() should trigger open testUrl handling in app when app is in foreground', async () => {
     await device.launchApp({newInstance: true});
-    await device.openURL({url: url});
-    await expect(element(by.text(url))).toBeVisible();
+    await device.openURL({url: testUrl});
+    await expect(element(by.text(testUrl))).toBeVisible();
   });
 
-  it('device.launchApp({url: url}) should trigger open url handling in app when app is in background', async () => {
-    const url = 'detoxtesturlscheme://such-string';
+  it('device.launchApp() with a URL should trigger url handling when app is in background', async () => {
     await device.launchApp({newInstance: true});
     await device.sendToHome();
-    await device.launchApp({newInstance: false, url: url});
-    await expect(element(by.text(url))).toBeVisible();
+    await device.launchApp({newInstance: false, url: testUrl});
+    await expect(element(by.text(testUrl))).toBeVisible();
+  });
+
+  it(':android: device.launchApp() with a URL should trigger url handling in activities running in single-task mode', async () => {
+    await device.launchApp({newInstance: true, androidSingleTask: true});
```

Comment:
Can't we detect task mode through the intent instead of asking the user to pass this metadata?

Reply:
The user would never do this. This is internal so that detox could rerun this specific test with its custom single-task activity.

Reply:
Come to think of it, perhaps I should do this duplication with the other url tests...

Reply:
err there are some typos there, I'll fix and force-push

Reply:
ok done with the dups.

Reply:
You're right, but I couldn't think of anything either...

Reply:
The `launchMode` cannot be set programatically

Reply:
Why not a unit test?